apps/issuer/serializers_v1.py

```diff
@@ -223,6 +225,122 @@ class BadgeClassExpirationSerializerV1(serializers.Serializer):
     duration = serializers.ChoiceField(source='expires_duration', allow_null=True,
                                        choices=BadgeClass.EXPIRES_DURATION_CHOICES)
 
+# https://stackoverflow.com/questions/50973569/django-rest-framework-relatedfield-cant-return-a-dict-object
+
+class ModifiedRelatedField(serializers.RelatedField):
+    def get_choices(self, cutoff=None):
+        queryset = self.get_queryset()
+        if queryset is None:
+            return {}
+
+        if cutoff is not None:
+            queryset = queryset[:cutoff]
+
+        return OrderedDict([
+            (
+                item.pk,
+                self.display_value(item)
+            )
+            for item in queryset
+        ])
+        
+class SuperBadgeBadgeClassField(ModifiedRelatedField):
+    def to_representation(self, value):
+        return {
+                "id": value.entity_id,
+                "name": value.name,
+                "description": value.description,
+                "image": value.image.url
+                }
+
+    def to_internal_value(self, data):
+        return BadgeClass.objects.get(entity_id=data)              
+
+class SuperBadgeClassSerializerV1(serializers.Serializer):
+    name = StripTagsCharField(required=True, max_length=128)
+    description = StripTagsCharField(required=False, allow_blank=True,
+            allow_null=True, max_length=255)
+    image = ValidImageField(required=False)
+
+    badges = SuperBadgeBadgeClassField(
+        queryset=BadgeClass.objects.all(), many=True, source='cached_badgeclasses'
+    )
+
+    def to_representation(self, instance):
+        representation = super(SuperBadgeClassSerializerV1, self).to_representation(instance)
+        return representation
+
+    def validate_image(self, image):
```

Comment:
This clearly doesn't just validate the image, but change the name and returns the image

Reply:
I copied the function from another place in the code. I would agree though that renaming the function would make sense, but i think then we should do it in all places

Reply:
Do you think it would make sense to do this in the scope of this PR?